### ISSUE TITLE
Add a boolean field that indicates potential sampling

### ISSUE THREAD

**Issue by pavel-shpilev** [uber-go/zap#686]
Now, this could be a controversial one, because of the added ambiguity.

There is no way to indicate that an entry has been sampled because of the very definition of sampling. But what can be done is a boolean field added to the _n_-th (where _n_ = `first`) and every subsequent entry within one tick. Let's call it `sampling_threshold_reached` for lack of better name.

This could be helpful for debugging. Say, when you see fewer messages of certain kind than what you'd expect. Such field could be a good hint to what's going on.

I'd love to hear whether this idea makes any sense and is worth looking into.

**Comment by prashantv** [comment]
We've discussed a similar feature in the past, maybe even a separate log line which tells you how many logs were sampled in some period.

**Comment by pavel-shpilev** [comment]
Yeah, a separate log line would be a better solution. I will start working on a special sampling core for that then.

**Comment by pavel-shpilev** [comment]
I'd love your thorough review on this, please.
https://github.com/uber-go/zap/pull/689

**Comment by pavel-shpilev** [comment]
> I'd love your thorough review on this, please.
> #689

Anyone? Pretty please

**Comment by pavel-shpilev** [comment]
Hi guys, I appreciate that everyone is busy, but could I PLEASE get some eyes on this pull request?
